libraries/AP_DroneCAN/AP_DroneCAN.cpp

```diff
@@ -1448,6 +1448,36 @@ void AP_DroneCAN::handle_ESC_status(const CanardRxTransfer& transfer, const uavc
 #endif
 }
 
+#if AP_EXTENDED_ESC_TELEM_ENABLED
+/*
+  handle Extended ESC status message
+ */
+void AP_DroneCAN::handle_esc_ext_status(const CanardRxTransfer& transfer, const uavcan_equipment_esc_StatusExtended& msg)
+{
+#if HAL_WITH_ESC_TELEM
```

Comment:
We should add a error somewhere if you get `AP_EXTENDED_ESC_TELEM_ENABLED` and not `HAL_WITH_ESC_TELEM`

Reply:
And then we won't need this check.

Reply:
Thank you @IamPete1, I have added the lines suggested into AP_ESC_Telem_config.h